app/com/linkedin/drelephant/exceptions/AnalyzedLog.java

```diff
@@ -0,0 +1,172 @@
+package com.linkedin.drelephant.exceptions;
+
+//Given a log string, analyzes and returns the relevant information!
+import org.apache.log4j.Logger;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class AnalyzedLog {
+
+        private static final Logger logger = Logger.getLogger(AnalyzedLog.class);
+        private String state;
+        private ExceptionLoggingEvent exception;
+        private Set<String> list;
```

Comment:
Fixed. It is a set of failed sub events. For instance, for Azkaban flow log returns the set of failed Azkaban job ids,
for Azkaban job log returns the set of failed MR jobs ids, for MR job log, returns the set of failed MR Task ids